### ISSUE TITLE
scoop update SSL Cert problem behind proxy

### ISSUE THREAD

**Issue by ankemp** [lukesampson/scoop#2337]
I'm having an issue with `scoop update` and `scoop status`, basically anything that uses git (I think).

When I run `scoop update`, I get:
```
Updating Scoop...
fatal: unable to access 'https://github.com/lukesampson/scoop/': SSL certificate problem: unable to get local issuer certificate
Scoop update failed.
```

The config setting I used was:
`scoop config proxy currentuser@sub.domain.com:8080`

Any idea how to fix this one?

**Comment by gitolicious** [comment]
Looks like your proxy terminates the SSL connection and replaces the certificate on the way. Follow https://stackoverflow.com/questions/11621768/how-can-i-make-git-accept-a-self-signed-certificate to accept this certificate.

**Comment by ankemp** [comment]
I'm not sure I want to do this globally, is there any way to set this specific to scoop?

**Comment by gitolicious** [comment]
Yes, it is also described in the [accepted answer](https://stackoverflow.com/a/11622001).
`cd c:\scoop\apps\scoop\current`
`git config http.sslVerify false`

**Comment by ankemp** [comment]
Maybe I'm doing something wrong?
``` bash
~/scoop/apps/scoop/current $ git config http.sslVerify false
fatal: not in a git directory
~/scoop/apps/scoop/current $
```

**Comment by gitolicious** [comment]
Not sure what console you are using there (doesn't look like cmd or PowerShell) but looks like something in your folder structure is different to mine... Can you try the commands
`scoop prefix scoop`
and
`scoop which scoop`
 and post the results?

**Comment by ankemp** [comment]
```bash
~ $ scoop prefix scoop
C:\Users\XXX\scoop\apps\scoop\current
~ $ scoop which scoop
C:\Users\XXX\scoop\apps\scoop\current\bin\scoop.ps1
```

**Comment by r15ch13** [comment]
This seems to be first-time `scoop update` is run. Scoops installation downloads the master.zip the first time and changes to a git repository after the first update.
You have to set config value globally, run scoop update, reset the global value and then change the local one.

**Comment by gitolicious** [comment]
Weird, that does look like a standard installation. Maybe you need to set it in the bucket directory?
```
cd c:\scoop\apps\scoop\current\bucket
git config http.sslVerify false`
```

**Comment by gitolicious** [comment]
Good catch @r15ch13 - I wasn't aware of that installation procedure...

**Comment by ankemp** [comment]
@r15ch13 & @gitolicious  Thanks for your help!

1) `git config --global http.sslVerify false`
2) `scoop update`
3) `git config --global http.sslVerify true`
4) `git config http.sslVerify false` in `scoop/apps/scoop/current/`
5) `scoop update` ... success!

**Comment by noloader** [comment]
@ankemp,

> git config --global http.sslVerify false

That's an awful answer. I want to puke when I see that Stack Overflow crap.

Run `git config -l` to get the location for `http.sslcainfo`. In my case, it was run from `C:\Users\<user id>\scoop\apps\scoop\current`. It will be something like `C:/Program Files/Git/mingw64/ssl/certs/ca-bundle.crt`. `ca-bundle.crt` is in PEM format. It is a concatenation of PEM encoded X.509 certificates. I.e., a bunch of `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----`.

Talk to your IT Guys, and get the company's Root CA certificates. You company may have a couple of stand-alone Root CA certificates. It may also have a Root CA certificate from an interception proxy, like NetSkope.

Add your company's PEM encoded certificates to the end of `ca-bundle.crt`. Git will use the new CA Root certificates to validate the chain.
